packages/lesswrong/lib/sql/Type.ts

```diff
@@ -196,24 +197,24 @@ const valueToString = (value: any, subtype?: Type): string => {
  * Annotate a type as having a default value. Subtype may or may not be concrete.
  */
 export class DefaultValueType extends Type {
-  constructor(private subtype: Type, private value: any) {
+  constructor(private type: Type, private value: any) {
     super();
   }
 
   toString() {
-    return `${this.subtype.toString()} DEFAULT ${this.valueToString()}`;
+    return `${this.type.toString()} DEFAULT ${this.valueToString()}`;
   }
 
   private valueToString() {
-    return valueToString(this.value, this.subtype);
+    return valueToString(this.value, this.type.isArray() ? this.type.subtype : undefined);
```

Comment:
This is where the bug was. DefaultValueType could have subtype ArrayType, in which case the actualy "subtype" we want to pass into valueToString is the subtype of the ArrayType.

The use of the word "subtype" here was making it more confusing so I've renamed them all to "type" apart from for ArrayType